samples/dnn/mask_rcnn.cpp

```diff
@@ -0,0 +1,233 @@
+// Deep learning based Object Detection and Instance Segmentation using Mask R-CNN 
+// VIDEO 360 DEMO: https://www.youtube.com/watch?v=0tU8991QgE8
+// Downloads:
+// http://download.tensorflow.org/models/object_detection/mask_rcnn_inception_v2_coco_2018_01_28.tar.gz
+// https://raw.githubusercontent.com/opencv/opencv_extra/master/testdata/dnn/mask_rcnn_inception_v2_coco_2018_01_28.pbtxt
+
+
+#include <fstream>
+#include <sstream>
+#include <iostream>
+#include <string.h>
+
+#include <opencv2/dnn.hpp>
+#include <opencv2/imgproc.hpp>
+#include <opencv2/highgui.hpp>
+
+#include "common.hpp"
+
+std::string keys =
+    "{ help  h     | | Print help message. \nUsage \n\t\t./mask_rcnn --image=logo.jpg \n\t\t ./mask_rcnn --media=teste.mp4}"
+    "{ image m     |<none>| Path to input image file.  }"
+    "{ video v     |<none>| Path to input video file.  }"
+    "{ cthr         | .5 | Confidence threshold. }"
+    "{ mthr        | .4 | Mask threshold. }";
+
+
+using namespace cv;
+using namespace dnn;
+using namespace std;
+
+float confThreshold;
+float maskThreshold;
+
+vector<string> classes;
+vector<Scalar> colors;
+
+void drawBox(Mat& frame, int classId, float conf, Rect box, Mat& objectMask);
+
+void postprocess(Mat& frame, const vector<Mat>& outs);
+
+int main(int argc, char** argv)
+{
+    CommandLineParser parser(argc, argv, keys);
+    parser.about("Use this script to run object detection using YOLO3 in OpenCV.");
+    if (argc == 1 || parser.has("help"))
+    {
+        parser.printMessage();
+        return 0;
+    }
+
+    confThreshold = parser.get<float>("cthr");
+    maskThreshold = parser.get<float>("mthr");
+
+    confThreshold = 0.5;
+    maskThreshold = 0.3;
+
+    string classesFile = "mscoco_labels.names";
+    ifstream ifs(classesFile.c_str());
+    string line;
+    while (getline(ifs, line)) classes.push_back(line);
+
+    string colorsFile = "colors.txt";
+    ifstream colorFptr(colorsFile.c_str());
+    while (getline(colorFptr, line)) {
+        char* pEnd;
+        double r, g, b;
+        r = strtod (line.c_str(), &pEnd);
+        g = strtod (pEnd, NULL);
+        b = strtod (pEnd, NULL);
+        colors.push_back(Scalar(r, g, b, 255.0));
+    }
+
+    String textGraph = "./mask_rcnn_inception_v2_coco_2018_01_28.pbtxt";
+    String modelWeights = "./mask_rcnn_inception_v2_coco_2018_01_28/frozen_inference_graph.pb";
+
+    Net net = readNetFromTensorflow(modelWeights, textGraph);
+    net.setPreferableBackend(DNN_BACKEND_OPENCV);
+    net.setPreferableTarget(DNN_TARGET_CPU);
+
+    string str, outputFile;
+    VideoCapture cap;
+    VideoWriter video;
+    Mat frame, blob;
+
+    try {
+
+        outputFile = "mask_rcnn_out_cpp.avi";
```

Comment:
Let's keep a sample as minimal as it possible. Just preview output image but not write an output files.

Reply:
Done